epub33/core/index.html

```diff
@@ -6082,9 +6082,112 @@ <h4>URLs in the OCF Abstract Container</h4>
-						the container, the string MUST be a <a data-cite="url#path-relative-scheme-less-url-string"
-							>path-relative-scheme-less-URL string</a>, optionally followed by <code>U+0023 (#)</code>
-						and a <a data-cite="url#url-fragment-string">URL-fragment string</a>. </p>
+						the container, the string SHOULD be a 
+						<a data-cite="url#path-relative-scheme-less-url-string">path-relative-scheme-less-URL string</a>, 
+						optionally followed by <code>U+0023 (#)</code> and a 
+						<a data-cite="url#url-fragment-string">URL-fragment string</a>. 
+					</p>
+
+					<p class="note">	
+						The properties of the <a>container root URL</a> are such that whatever the amount
+						of <a data-cite="url/#double-dot-path-segment">double-dot path segments</a> in a URL string (for
+						example, <code>../../../secret</code>), it is parsed to a <a>content URL</a> (and not "leak"
+						outside the container). This avoids potential run-time security issues. 
+						Furthermore, the additional constraint and <a href="#algo-out-of-container">algorithm</a> below 
+						ensure that such potentially problematic URLs can also be detected when checking the EPUB Document.
+					</p>
+
+					<p>For a URL string <var>url</var> found in the <a>OCF Abstract Container</a>, the following steps MUST also return <var>true</var>:</p>
+
+					<ol class="algorithm" id="algo-out-of-container">
+						<li>
+							Set the <a>container root URL</a> to <code>https://a.example.org/A/</code>.
+							<details>
+								<summary>Explanation</summary>
+								<p class="note">
+									The goal of the algorithm is to detect whether <var>url</var> could be seen as
+									"leaking" outside the container. 
+									To do that, the official <a data-cite="url#concept-url-parser">URL parsing algorithm</a> 
+									is used with an artificial root URL; the detection of the "leak" is done by 
+									comparing the result of the parsing with the presence of the first test path 
+									segment (<code>A</code>). 
+									(Note that the artificial container root URL wilfully violates, for the purpose of this
+									algorithm, the <a href="#confreq-root-url">required properties</a> by using 
+									that first test path segment.)     
+								</p>
+							</details>
+						</li>
+
+						<li>
+							Let <var>base</var> be the <a data-cite="url#concept-base-url">base URL</a> that must be used to parse <var>url</var> as defined by the context (document or environment) where <var>url</var> is used, and according to the <a>content URL</a> of the <a>Package Document</a> (see <a href="#sec-parse-package-urls"></a>). 
+							<details>
+								<summary>Explanation</summary>
+								<p class="note">
+									In the case of a URL in the package document the <var>base</var> variable is set 
+									to the <a>container root URL</a>. In the case of a URL in an XHTML Content Document, the 
```

Comment:
Copying a previous comment, that was left out:

s/container root URL/content URL of the Package Document/

Also, we could link to section "Parsing URLs in the Package Document" so that the reader can navigate to where it is exactly defined.

Reply:
On the extra link: it is already there in the 'standard' part of the text, I did not want to make the note too heavy.